modules/ensemble/lib/widget/loading_container.dart

```diff
@@ -105,6 +107,45 @@ class LoadingContainerState extends WidgetState<LoadingContainer> {
         : loadingWidget ?? const SizedBox.shrink();
   }
 
+  LinearGradient _buildGradient() {
+    switch (widget._controller.shimmerEffect) {
+      case ShimmerEffect.horizontal:
+        return LinearGradient(
+          begin: Alignment.centerLeft,
+          end: Alignment.centerRight,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+      case ShimmerEffect.vertical:
+        return LinearGradient(
+          begin: Alignment.topCenter,
+          end: Alignment.bottomCenter,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+      case ShimmerEffect.diagonal:
+      default:
+        return LinearGradient(
+          begin: Alignment.topLeft,
+          end: Alignment.centerRight,
+          colors: _buildGradientColors(),
+          stops: const [0.1, 0.3, 0.4],
+          tileMode: TileMode.clamp,
+        );
+    }
+  }
+
+  List<Color> _buildGradientColors() {
+    return <Color>[
```

Comment:
@hemish11 please create a separate ticket to expose these colors as style properties so people could add them to theme etc. Right now they are hardcoded. It's a separate ticket